packages/hint-compat-api/src/helpers/compat-html.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @fileoverview Helper that contains all the logic related with HTML compat api, to use in different modules.
+ */
+import { HTMLParse, HTMLEvents } from '@hint/parser-html/dist/src/types';
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { ElementFound, IAsyncHTMLElement, ProblemLocation, AsyncHTMLAttribute, IAsyncNamedNodeMap } from 'hint/dist/src/lib/types';
+
+import { MDNTreeFilteredByBrowsers, TestFeatureFunction, FeatureInfo } from '../types';
+import { CompatStatement } from '../types-mdn.temp';
+import { CompatBase } from './compat-base';
+
+export class CompatHTML extends CompatBase<HTMLEvents, HTMLParse> {
+    public constructor(hintContext: HintContext<HTMLEvents>, MDNData: MDNTreeFilteredByBrowsers, testFunction: TestFeatureFunction) {
+        super(hintContext, MDNData, testFunction);
+
+        hintContext.on('parse::end::html', this.onParse.bind(this));
```

Comment:
Ok, I just updated everything to remove the `html-parser` dependency. As expected, a bunch of tests are failing because the missing location issue. Should I apply any workaround to handle this situation or wait for your changes in master?

Reply:
I dig some digging and realized the helper `context.report` uses to resolve a location given an `element` is, in fact, public. That made this a pretty straightforward fix to swap out the current `element.getLocation` call with `hintContext.findProblemLocation(element)` (which uses `getLocation` internally).

I validated this fixed the tests (though I also found and fixed an issue with how one of the other helpers was being used) and updated the PR with the fix.

Reply:
👍